### PR TITLE
C builder image in trunk

### PR DESCRIPTION
**Pull request by sjmiller609**
Create a new builder image for building C extensions, which is also a subset of the PGX builder image.

### CODE REVIEW

File: .github/workflows/images.yml [hunk 1]
```diff
@@ -77,6 +77,8 @@ jobs:
           # Tag with 'latest'
           # if the branch is 'main'
           publish_latest: ${{ steps.tags.outputs.tag_latest }}
+          # If we are publishing latest, also tag it with calver
+          publish_calver: ${{ steps.tags_outputs.tag_latest }}
```
Comment (sjmiller609) [inline .github/workflows/images.yml:81]: For any push to a branch changing a docker image, build and push the hash. If it's main, then also push latest and calendar versions.

File: trunk/cli/images/pgx-builder/Dockerfile [hunk 2]
```diff
@@ -1,22 +1,4 @@
-ARG UBUNTU_VERSION=22.04
```
Comment (sjmiller609) [inline trunk/cli/images/pgx-builder/Dockerfile:1]: the build environment for C extensions seems like it's a subset of the build environment for PGX

File: .github/workflows/trunk-cli.yml [hunk 3]
```diff
@@ -0,0 +1,77 @@
+name: Trunk CLI workflow
```
Comment (sjmiller609) [inline .github/workflows/trunk-cli.yml:1]: This workflow runs linting, unit test, and a functional test that makes sure pgx-builder will build a pgx extension - adding so I make sure I do not break pgx builder while working on C builder.

**Review by ChuckHend — APPROVED** [review]

File: trunk/cli/tests/integration_tests.rs [hunk 4]
```diff
@@ -15,15 +15,6 @@ fn help() -> Result<(), Box<dyn std::error::Error>> {
...
 }
 
-#[test]
-fn install() -> Result<(), Box<dyn std::error::Error>> {
-    let mut cmd = Command::cargo_bin(CARGO_BIN)?;
-    cmd.arg("install");
-    cmd.assert()
-        .stdout(predicate::str::contains("not implemented"));
-    Ok(())
-}
```
Comment (sjmiller609) [inline trunk/cli/tests/integration_tests.rs:25]: trunk install is now implemented, so we can delete this test

**Pull request merged**